test/QueryToSocketDistributorTest.cpp

```diff
@@ -146,7 +149,8 @@ ASYNC_TEST(QueryToSocketDistributor, listeningBeforeStartWorks) {
 // _____________________________________________________________________________
 
 ASYNC_TEST(QueryToSocketDistributor, addQueryStatusUpdateBeforeListenersWorks) {
-  QueryToSocketDistributor queryToSocketDistributor{ioContext, noop};
+  auto ptr = std::make_shared<QueryToSocketDistributor>(ioContext, noop);
+  QueryToSocketDistributor& queryToSocketDistributor = *ptr;
```

Comment:
I'm still convinced changing this to a single variable, to avoid the `ptr` dummy variable would be the nicer approach here, even if it makes the diff larger. But no strong opinion on this.